internal/dependencies/tools/validator.go

```diff
@@ -127,6 +131,16 @@ func (tv *Validator) validateTools(i any, kfdManifest config.KFD) ([]string, []e
 		}
 
 		tool := tv.toolFactory.Create(itool.Name(toolName), toolCfg.Version)
+
+		if tool == nil {
+			errs = append(
+				errs,
+				fmt.Errorf("%w: %s", ErrToolNotFound, toolName),
```

Comment:
I think that the order should be inverted for better readability and the version that we are checking included in the error message too

```suggestion
				fmt.Errorf("%s version %s: %w", toolName, ErrToolNotFound),
```

Reply:
Let me check :)